plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMaterializedView.java

```diff
@@ -55,6 +88,29 @@ protected DistributedQueryRunner createQueryRunner()
                     .setCatalog("iceberg2")
                     .build();
 
+            RandomTableFunctionProcessorProvider processorProvider = new RandomTableFunctionProcessorProvider();
+            queryRunner.installPlugin(new MockConnectorPlugin(MockConnectorFactory.builder()
+                    .withTableFunctions(ImmutableSet.of(new RandomTableFunction()))
+                    .withFunctionProvider(Optional.of(new FunctionProvider()
+                    {
+                        @Override
+                        public TableFunctionProcessorProvider getTableFunctionProcessorProvider(ConnectorTableFunctionHandle functionHandle)
+                        {
+                            if (functionHandle instanceof RandomTableFunctionHandle) {
```

Comment:
What is special about RandomTableFunctionHandle?
Does it matter that the function is random?

BTW you can also test without adding a new function, using `table_changes`

Reply:
I thought about `table_changes` but that seemed to be an overkill for this, this one is also quite complex but at least tests themselves are simpler.
The function doesn't have to return random result, it can return a sequence etc. as long as each call returns different value. 
I saw your other comment and I will switch to return sequence